src/huak/env/venv.rs

```diff
@@ -52,22 +69,59 @@ impl Venv {
 
         Ok(name)
     }
-}
 
-impl Default for Venv {
-    fn default() -> Venv {
-        let cwd = match env::current_dir() {
-            Err(_) => Path::new(".").to_path_buf(),
-            Ok(p) => p,
-        };
-
-        Venv {
-            path: cwd.join(DEFAULT_VENV_NAME),
+    /// Activates the virtual environment in the current shell
+    pub fn activate(&self) -> HuakResult<()> {
+        // Check if venv is already activated
+        if env::var(HUAK_VENV_ENV_VAR).is_ok() {
+            return Err(HuakError::VenvActive);
         }
+
+        let script = self.get_activation_script()?;
+        let source_command = get_shell_source_command()?;
+        let activation_command =
+            format!("{} {}", source_command, script.display());
+
+        env::set_var(HUAK_VENV_ENV_VAR, "1");
+
+        // Spawn a pseudo-terminal with current shell and source activation script
+        let shell_path = get_shell_path()?;
+        let mut new_shell = expectrl::spawn(&shell_path)?;
+        let mut stdin = expectrl::stream::stdin::Stdin::open()?;
+        new_shell.send_line(&activation_command)?;
+        new_shell.set_expect_lazy(false);
+        let (cols, rows) = terminal_size::terminal_size().unwrap();
```

Comment:
I'm fairly new to rust, so correct me if this is misinformed.

We're trying to refrain from using unwrap in lib or application code. We use unwrap in tests, but everywhere else we're trying to be explicit about error handling. Can we map an error to this or default?

Open to thoughts here as well.

Reply:
You're totally right, I just missed that one unwrap (:
I think best way to handle this is to silently omit following `set_window_size`. Then expectrl will use it's default. Once we add logging we should log this somewhere tho.